# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/printing/nsPrintJob.cpp
# Commit: 4e05361c6b08
# Full Hash: 4e05361c6b08aad43b939c33f5aa608a3d9ccbc8
# Author: Jonathan Watt <jwatt@jwatt.org>
# Date: 2022-05-23 09:52:44
# Regressor Bug: 1770539
# File Overlap Count: 1
# Description:
#   Bug 1770539 p5 - Move nsPrintData::mPreparingForPrint to nsPrintJob. r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D146977
# ==============================================================================

diff -r cc4de7b25404 -r 4e05361c6b08 layout/printing/nsPrintJob.cpp
--- a/layout/printing/nsPrintJob.cpp	Sun May 22 09:37:56 2022 +0000
+++ b/layout/printing/nsPrintJob.cpp	Sun May 22 09:37:57 2022 +0000
@@ -230,9 +230,7 @@
   DisconnectPagePrintTimer();
 }
 
-bool nsPrintJob::CheckBeforeDestroy() const {
-  return mPrt && mPrt->mPreparingForPrint;
-}
+bool nsPrintJob::CheckBeforeDestroy() const { return mPreparingForPrint; }
 
 //-------------------------------------------------------
 void nsPrintJob::Destroy() {
@@ -1667,7 +1665,7 @@
     MOZ_ASSERT(seqFrame, "no page sequence frame");
 
     // We are done preparing for printing, so we can turn this off
-    printData->mPreparingForPrint = false;
+    mPreparingForPrint = false;
 
 #ifdef EXTENDED_DEBUG_PRINTING
     nsIFrame* rootFrame = poPresShell->GetRootFrame();
@@ -1865,8 +1863,8 @@
 //---------------------------------------------------------------------
 void nsPrintJob::SetIsPrinting(bool aIsPrinting) {
   mIsDoingPrinting = aIsPrinting;
-  if (mPrt && aIsPrinting) {
-    mPrt->mPreparingForPrint = true;
+  if (aIsPrinting) {
+    mPreparingForPrint = true;
   }
 }
 